src/Taweret/mix/trees.py

```diff
@@ -625,37 +636,33 @@ def plot_prediction(self, xdim: int = 0):
         the predict function.
 
         Parameters:
-        ----------
-        :param int xdim: index of the column to plot against the predictions.
+        -----------
+        :param int xdim: index of the column to plot against the
+                predictions.
 
         Returns:
         --------
         :return: None.
         '''
         col_list = ['red', 'blue', 'green', 'purple', 'orange']
-        if self.pred_mean is None:
-            # Compute weights at training points
-            print("Getting predictions at training points by default.")
-            out_pred = self.predict(self.X_train.transpose())
-            self.X_test = self.X_train.transpose()
-
-        # Now plot the prediction -- need to improve this plot
-        fig = plt.figure(figsize=(6, 5))
-        plt.plot(self.X_test[:, xdim], self.pred_mean, color='black')
-        plt.plot(self.X_test[:, xdim], self.pred_lower,
-                 color='black', linestyle="dashed")
-        plt.plot(self.X_test[:, xdim], self.pred_upper,
-                 color='black', linestyle="dashed")
-        for i in range(self.nummodels):
-            plt.plot(self.X_test[:, xdim], self.F_test[:, i],
-                     color=col_list[i], linestyle='dotted')
-        # Recall X_train was transposed in the beginning
-        plt.scatter(self.X_train[xdim, :], self.y_train)
-        plt.title("Posterior Mean Prediction")
-        plt.xlabel("X")  # Update Label
-        plt.ylabel("F(X)")  # Update Label
-        plt.grid(True, color='lightgrey')
-        plt.show()
+        if self.pred_mean is not None:
```

Comment:
@jcyannotty You've changed the logic at this line and in its code block.  Is that intentional?  Was it a bug?  If so, does that mean that the test suite didn't detect the bug?

Reply:
I changed the logic because of .flake8. So no, it was not a bug at all, I just needed to find a way to satisfy `./src/Taweret/mix/trees.py:654:13: F841 local variable 'out_pred' is assigned to but never used`
 and `./src/Taweret/mix/trees.py:696:13: F841 local variable 'out_wts' is assigned to but never used`. In fact, these are simply plotting functions, which have no impact on the modeling. They are just convenient utilities for the user. I think I actually remove them all together in new code base.